torchtext/experimental/models/roberta_model.py

```diff
@@ -0,0 +1,74 @@
+import os
+import torch
+import torch.nn as nn
+from torchtext.experimental.modules import TransformerEncoder
+from .roberta_transform import RobertaTransform
+from torchtext.utils import download_from_url, extract_archive, files_exist, load_args_from_json
+
+
+class RobertaModel(nn.Module):
```

Comment:
nit: I don't see "Model" suffixes on torchvision models; I suspect we'll prefer `models.Roberta` instead of `models.RobertaModel`?
Do we care about matching the capitalization/spelling from the paper (I see it as `RoBERTa` here: https://arxiv.org/abs/1907.11692)

Reply:
This idea is still prototype. My plan is to have XXModel as a wrapper of pretrained models, preprocessing transforms and config. So RoBERTa (without the "model" suffix) will stay in the module folder. But this is still under discussions with the broader audience.

Reply:
@zhangguanheng66 could you add more variations to demo how to reuse the building blocks and the inheritance between them? e.g. RobertaModel, RobertaModelForMaskedLM,  RobertaModelForClassification, XlmrModel*, etc. Do we need define to Head class for each variation?

Reply:
I would vote for removing the `Model` suffix.

I wouldn't have classes for wrapping pre-trained models, but instead builder functions.

So instead of having a `RoBERTa` (generic class` and a `RoBERTaModel` (which loads pre-trained weights and sets up the right hyperparameters), I would instead have a `RoBERTa` for the generic class and a function `roberta_typeXx_ondataset_y` for building the model

Reply:
> I would instead have a `RoBERTa` for the generic class and a function `roberta_typeXx_ondataset_y` for building the model

Would the generic class here possibly just be `BERT`?  It seems BERT is the architecture of the model and RoBERTa is a method for training it?

Reply:
> Would the generic class here possibly just be BERT? It seems BERT is the architecture of the model and RoBERTa is a method for training it?

Possibly, I don't know the specific details of RoBERTa to be 100% sure, but that's something worth considering if possible

Reply:
I put the generic `TransformerEncoder` class (similar to torchvision's [ResNet](https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py#L144)) in the `modules` folder, which is composed of embedding layer plus the transformer encoder. Here, I can have a class `RobertaModel` to: 1) create an instance of the `TransformerEncoder` class 2) load pretrained weights 3) load the preprocessing transforms. Of course, this class can be replaced with a factory func, like [resnet18](https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py#L268).